internal/backends/winit/accesskit.rs

```diff
@@ -0,0 +1,437 @@
+        }
+
+        let min = item.accessible_string_property(AccessibleStringProperty::ValueMinimum);
+        let max = item.accessible_string_property(AccessibleStringProperty::ValueMaximum);
+        let step = item.accessible_string_property(AccessibleStringProperty::ValueStep);
+        let value = item.accessible_string_property(AccessibleStringProperty::Value).to_string();
+
+        match (min.parse(), max.parse(), value.parse(), step.parse()) {
+            (Ok(min), Ok(max), Ok(value), Ok(step)) => {
+                builder.set_min_numeric_value(min);
+                builder.set_max_numeric_value(max);
+                builder.set_numeric_value(value);
+                builder.set_numeric_value_step(step);
+            }
+            _ => {
+                builder.set_value(value);
+            }
+        }
+
+        builder
+    }
+}
+
+// Encode a u128 accesskit node id for an ItemRc by assuming that the ItemRc remains
+// alive and storing the item index, raw VRc inner pointer, and a tree generation index
+// in the u128.
+fn node_id_for_item_rc(mut item: ItemRc, tree_generation: u16) -> NodeId {
+    while !item.is_accessible() {
+        if let Some(parent) = item.parent_item() {
+            item = parent;
+        } else {
+            break;
+        }
+    }
+
+    let component_ptr = item.component().as_raw_ptr();
+    let component_usize: u128 = component_ptr as usize as u128;
+    let item_index: u128 = item.index() as u32 as u128;
+
+    let encoded_id = std::num::NonZeroU128::new(
+        (component_usize << 64) | ((tree_generation as u128) << 48) | item_index,
+    )
+    .unwrap();
+    NodeId(encoded_id)
+}
+
+fn item_rc_for_node_id(id: NodeId, expected_generation: u16) -> Option<ItemRc> {
+    let encoded_component = id.0.get() >> 64;
+    let encoded_generation = ((id.0.get() >> 48) & 0xffff) as u16;
+    let encoded_item_index = (id.0.get() & 0xffffffff) as u32;
+
+    if encoded_generation != expected_generation {
+        return None;
+    }
+
+    // Safety: When the tree structure changes, the tree generation value is increased, causing a mismatch
+    // caught earlier. Therefore, at this point the pointer should be safe to convert back into a VRc.
+    let component_rc = unsafe {
+        vtable::VRc::<ComponentVTable, vtable::Dyn>::from_raw_ptr(encoded_component as *const _)
+    };
+
+    Some(ItemRc::new(component_rc, encoded_item_index as usize))
+}
+
+struct AccessibilitiesPropertyTracker {
+    window_adapter_weak: Weak<WinitWindowAdapter>,
+}
+
+impl i_slint_core::properties::PropertyDirtyHandler for AccessibilitiesPropertyTracker {
+    fn notify(&self) {
+        let win = self.window_adapter_weak.clone();
+        i_slint_core::timers::Timer::single_shot(Default::default(), move || {
+            if let Some(window_adapter) = win.upgrade() {
+                window_adapter.accesskit_adapter.rebuild_tree_of_dirty_nodes();
+            };
+        })
+    }
+}
+
+struct CachedNode {
```

Comment:
Why don't you put the ItemRc in this structure. 
Then the NodeId could just be an index in the all_nodes vector. (maybe with a generation attached to it)
That way you don't need these "highly unsafe" trick :-)

(Maybe you nead an ItemWeak actually)

Reply:
Yeah, now that you mention it...

ItemWeak is bigger than u128 but that's okay.

I'll give it a try :)

Reply:
This worked well in the end, just using the index in the nodes vec as node id (mangled with the generation, of course).